warehouse/templates/packaging/detail.html

```diff
@@ -77,12 +77,15 @@ <h3 class="package-snippet__title">
   {%- if test_environment -%}
     {% set index_url = " -i https://test.pypi.org/simple/" %}
   {%- endif -%}
-
   {%- if request_route == "packaging.release" -%}
     {% set project_version = "==" + release.version %}
   {%- endif -%}
 
-  pip install{{ index_url }} {{ release.project.name }}{{ project_version }}
+  {%- if '!' in release.version -%}
```

Comment:
The `Version` class has an [`epoch` property https://github.com/pypa/packaging/blob/2a883ca86300bab76ad6296860797907b3432871/packaging/version.py#L348-L352] that can be used for this instead.